src/Http/ProxyClient.php

```diff
@@ -4,86 +4,55 @@
 namespace Eggheads\CakephpCommon\Http;
 
 use Cake\Core\Configure;
-use Psr\Http\Message\RequestInterface;
-use Cake\Http\Client\Response;
+use Eggheads\CakephpCommon\Error\InternalException;
 
 /**
  * Класс для работы с прокси
  *
- * Настройка:
- * - Прокси берутся из базы данных
- * - Имя подключения можно указать в конфигурации с ключом proxyDBConfig (по-умолчанию default)
- * - Имя таблицы с прокси можно указать в конфигурации с ключом proxyTableName (по-умолчанию proxy_config)
- *
- * Метод получения конфига прокси находится в классе
- * @see ProxyList
- *
  * Использование:
  * Можно отключить использование прокси, добавив параметр конфигурации 'isProxyEnabled' => false
  * В этом случае будет работать как обычный \Eggheads\CakephpCommon\Http\Client
- *
- * При ошибке запроса Curl перечисленной в \Eggheads\CakephpCommon\Http\Client::REPEAT_REQUEST_CURL_ERROR,
- * происходит автоматическая смена прокси.
- * Данное поведение можно отключить вызвав метод
- * @see ProxyClient::doNotChangeProxyAfterError()
- *
- * Для смены прокси между запросами, можно вызвать метод
- * @see ProxyClient::changeProxy()
  */
 class ProxyClient extends Client
 {
-    /** @var bool Менять ли прокси после ошибки curl */
-    private bool $_isChangeProxyAfterError = true;
+    /** @var string Элемент массива с параметрами в общем конфиге */
+    public const CONFIG_FIELD_NAME = 'proxyConfig';
 
     /**
      * @inheritDoc
      * @phpstan-ignore-next-line
+     * @throws InternalException
      */
     public function __construct(array $config = [])
     {
-        $this->changeProxy();
+        if (Configure::read('isProxyEnabled', true)) {
+            $this->setConfig('proxy', $this->_getConfig());
+        }
+
         parent::__construct($config);
     }
 
     /**
-     * Меняем прокси при ошибке запроса
+     * Получение и проверка конфигурации
      *
-     * @inheritDoc
+     * @return array<string, string> ['proxy' => 'string','username' => 'string','password' => 'string']
```

Comment:
Корректнее использовать array shapes.
```suggestion
     * @return array{proxy: string, username: string, password: string}
```